### PR TITLE
[Fix] binding obviously wrong issuses by X64 & MSVC

### PR DESCRIPTION
**Pull request by TrustHenry**
X64 system has corrected some errors found while using MSVC to link.

Related to #264

### CODE REVIEW

File: source/agora/utils/Log.d [hunk 1]
```diff
@@ -261,7 +261,8 @@ public class AgoraLayout : Appender.Layout
 
 ***************************************************************************/
 
-private extern (C++) void writeDLog (const char* logger, int level, const char* msg)
+private extern (C++) void writeDLog (const (char)* logger, int level,
+    const (char)* msg)
```
Comment (Geod24) [inline source/agora/utils/Log.d]:
```suggestion
private extern(C++) void writeDLog (const(char)* logger, int level,
    const(char)* msg)
```

File: source/scpd/scp/QuorumSetUtils.d [hunk 2]
```diff
@@ -22,4 +22,4 @@ bool isQuorumSetSane(ref const SCPQuorumSet qSet, bool extraChecks,
     const(char)** reason = null);
 
 // normalize the quorum set, optionally removing idToRemove
-void normalizeQSet(ref SCPQuorumSet qSet, const NodeID* idToRemove = null);
+void normalizeQSet(ref SCPQuorumSet qSet, const (NodeID)* idToRemove = null);
```
Comment (Geod24) [inline source/scpd/scp/QuorumSetUtils.d]:
```suggestion
void normalizeQSet(ref SCPQuorumSet qSet, const(NodeID)* idToRemove = null);
```

File: source/scpd/tests/LayoutTest.d [hunk 3]
```diff
@@ -33,11 +33,11 @@ extern(C++) struct FieldInfo
 
 static foreach (Type; GlueTypes)
 {
-    extern(C++) FieldInfo cppFieldInfo (ref Type, const(char)*);
+    extern(C++) FieldInfo cppFieldInfo (ref Type, const(char*));
```
Comment (Geod24) [inline source/scpd/tests/LayoutTest.d]: Looking at the C++ definition this is wrong

**Review by Geod24 — CHANGES_REQUESTED** [review]

**Comment by Geod24** [comment]
@TrustHenry : When writing commits message, please try to make them more expressive.
E.g. instead of "MSVC mangle them differently", say "Add extern(C++, class) to some D bindings".
The commit title should be a very brief (~50 chars max) description of what is being (another frequent mistake: It is not how things are being done, but what is being done).

The "why" usually go in the commit message, not the title.

File: source/scpd/tests/LayoutTest.d [hunk 4]
```diff
@@ -24,7 +24,7 @@ import std.string;
 import std.traits;
 
 /// Contains the size and offset of a field within a struct
-extern(C++) struct FieldInfo
+extern(C++, class) struct FieldInfo
```
Comment (Geod24) [inline source/scpd/tests/LayoutTest.d]: https://github.com/bpfkorea/agora/blob/d8154e24ed91844e7730ed646b5c3913cb5ec8fc/source/scpp/extra/DLayoutChecks.cpp#L31
???

**Comment by TrustHenry** [comment]
@Geod24 I've applied all the suggestions.
Modified to a more expressive commit message.

File: source/scpd/tests/LayoutTest.d (hunk 4 above)
Comment (AndrejMitrovic) [reply source/scpd/tests/LayoutTest.d]: This is a struct, please revert this change..

Comment (TrustHenry) [reply source/scpd/tests/LayoutTest.d]: It's my mistake.
I don't know why the changes keep going back. I don't think I'm familiar with Git.

Comment (AndrejMitrovic) [reply source/scpd/tests/LayoutTest.d]: Which commands did you run in git?

Comment (TrustHenry) [reply source/scpd/tests/LayoutTest.d]: I used git rebase frequently.
Maybe I used the wrong rebase in the middle

**Review by Geod24 — APPROVED** [review]
Merge at will

**Comment by TrustHenry** [comment]
@AndrejMitrovic If it looks good, can you merge it?

**Pull request merged**